### PR TITLE
UI bug fixes ( Sync EE to OSS )

### PR DESCRIPTION
**Pull request by rameshmane7218**
## Change Summary

- #7301
- #7353 
- #7354

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/grid/GroupBy.vue [hunk 1]
```diff
@@ -154,18 +153,21 @@ const parseKey = (group: Group) => {
...
-    return [dayjs(key).utc().local().format(dateTimeFormat)]
+  if (key && group.column?.uidt === UITypes.DateTime) {
+    const dateTime = parseStringDateTime(
+      key,
+      `${parseProp(group.column?.meta)?.date_format ?? dateFormats[0]} ${
+        parseProp(group.column?.meta)?.time_format ?? timeFormats[0]
+      }`,
+    )
+
+    return [dateTime]
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/grid/GroupBy.vue:164]: `dateTime` is not used afterwards. you may directly return it.

```ts
return [
  parseStringDateTime(
    key,
    `${parseProp(group.column?.meta)?.date_format ?? dateFormats[0]} ${
      parseProp(group.column?.meta)?.time_format ?? timeFormats[0]
    }`,
  )
]
```

File: packages/nc-gui/components/smartsheet/grid/GroupBy.vue [hunk 2]
```diff
@@ -154,18 +153,21 @@ const parseKey = (group: Group) => {
...
   }
 
   // show the groupBy time field title format as like cell format
-  if (key && group.column?.uidt === UITypes.Time && dayjs(key).isValid()) {
-    return [dayjs(key).format(timeFormats[0])]
+  if (key && group.column?.uidt === UITypes.Time) {
+    const time = parseStringDateTime(key, timeFormats[0], false)
+    return [time]
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/grid/GroupBy.vue:170]: same as above

Comment (rameshmane7218) [reply packages/nc-gui/components/smartsheet/grid/GroupBy.vue:170]: Okay, I will remove this and from other place `Links` -> `ListItem` and related computed values `displayValueTypeAndFormatProp` & `headerDisplayValue` from `useLTARStore.ts` file

Comment (rameshmane7218) [reply packages/nc-gui/components/smartsheet/grid/GroupBy.vue]: done

File: packages/nc-gui/components/smartsheet/grid/GroupBy.vue (hunk 1 above)
Comment (rameshmane7218) [reply packages/nc-gui/components/smartsheet/grid/GroupBy.vue]: done

**Pull request merged**